core/src/main/java/org/openstack4j/openstack/networking/domain/NeutronLayer2Gateway.java

```diff
@@ -0,0 +1,235 @@
+    }
+
+    @Override
+    public Layer2GatewayBuilder toBuilder() {
+        return new Layer2GatewayConcreteBuilder(this);
+    }
+
+    public static class Gateways extends ListResult<NeutronLayer2Gateway> {
+
+        private static final long serialVersionUID = 1L;
+
+        @JsonProperty("gateways")
+        private List<NeutronLayer2Gateway> gateways;
+
+        @Override
+        protected List<NeutronLayer2Gateway> value() {
+            return gateways;
+        }
+    }
+
+    public static class DeviceImpl implements Device {
+
+        public DeviceImpl(){}
+
+        @JsonCreator
+        public DeviceImpl(
+                @JsonProperty("interface_names") String interfaceNames,
+                @JsonProperty("name") String deviceName
+        ) {
+            this.deviceName = deviceName;
+            if (interfaceNames != null) {
+                this.interfaces = Arrays.stream(interfaceNames.split(";"))
+                        .map(InterfaceImpl::new).collect(Collectors.toList());
+            }
+        }
+
+        @JsonProperty("interfaces")
+        private List<Interface> interfaces;
+
+        @JsonProperty("device_name")
+        private String deviceName;
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getDeviceName() {
+            return deviceName;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public List<Interface> getInterfaces() {
+            return interfaces;
+        }
+    }
+
+    public static class InterfaceImpl implements Device.Interface {
+
+        @JsonProperty("name")
+        private String name;
+        @JsonProperty("segmentation_id")
+        private List<String> segmentationId;
+
+        public InterfaceImpl(){}
+
+        public InterfaceImpl(String interfaceName) {
+            if (interfaceName.contains("|")) {
+                this.name = interfaceName.split("\\|")[0];
+                this.segmentationId = Arrays.asList(interfaceName.split("\\|")[1].split("#"));
+            } else {
+                this.name = interfaceName;
+            }
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public List<String> getSegmentationId() {
+            return segmentationId;
+        }
+    }
+
+    public static class Layer2GatewayConcreteBuilder extends ResourceBuilder<Layer2Gateway, Layer2GatewayConcreteBuilder> implements Layer2GatewayBuilder {
+        private NeutronLayer2Gateway m;
+
+        public Layer2GatewayConcreteBuilder(NeutronLayer2Gateway policy) {
+            this.m = policy;
+        }
+
+        public Layer2GatewayConcreteBuilder() {
+            this.m = new NeutronLayer2Gateway();
+        }
+
+        @Override
+        protected NeutronLayer2Gateway reference() {
+            return m;
```

Comment:
You can reference `m` without using `this.`?
I feel like I might have asked this before.

Reply:
You can as long as there's not a local variable `m` as well.
```
private Foo m;
Foo setM(Foo m){
    m = m; // won't modify the class field
    this.m = m; // works
}
```

Reply:
Thanks. I'm not going to promise to not ask the same question again, but I'll try. :-)